designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/DBFetchingProcessRepository.scala

```diff
@@ -77,9 +77,10 @@ abstract class DBFetchingProcessRepository[F[_]: Monad](
           .getLastActionPerProcess(ProcessActionState.FinishedStates, Some(ScenarioActionName.StateActions))
       )
       // for last deploy action we are not interested in ExecutionFinished deploys - we don't want to show them in the history
-      lastDeployedActionPerProcess <- fetchActionsOrEmpty(
-        actionRepository.getLastActionPerProcess(Set(ProcessActionState.Finished), Some(Set(ScenarioActionName.Deploy)))
-      )
+      lastDeployedActionPerProcess = lastActionPerProcess.filter { case (_, action) =>
+        action.actionName == ScenarioActionName.Deploy && action.state == ProcessActionState.Finished
```

Comment:
Why is it better?  After that we won't see deploy actions if there are some actions after it

Reply:
>After that we won't see deploy actions if there are some actions after it
that was already the case

see the PR description, the only thing that changed was it made to behave the same as `fetchProcessDetailsForVersion`

Reply:
i do think this code is still ugly, not very readable, and potentially just plain wrong, but the objective of this quick PR was to at least make it consistent

i've got a bigger fix (and performance improvement) cooking in a separate PR (that is probably too large to be backported to 1.15)

Reply:
Why are you sure that `fetchProcessDetailsForVersion` shouldn't behave like `fetchLatestProcessDetailsByQueryAction` ?

Reply:
ok, perhaps this requires more analysis, here's how i understand the intended 'meaning' of the fields

- '`lastActionData`' - latest Finished/ExecutionFinished action (could perhaps be relaxed to latest action regardless of ProcessActionState), both methods behave the same

- '`lastStateActionData`' - latest Finished/ExecutionFinished state action, so last action that succeeded and impacted the process state (Cancel, Deploy, Pause) - IMO `fetchProcessDetailsForVersion` behaves correctly

- '`lastDeployedActionData`' (the tricky one) - latest Finished deploy action, that hasn't yet been ExecutionFinished, so if a process has such an action, it means that it is currently deployed (and this 'meaning' is used by `fetchLatestProcessDetailsByQueryAction -> fetchLatestProcessesQuery`) - IMO both behave incorrectly

Reply:
so if we want to achieve the (in my opinion) desired outcome for `lastDeployedActionData` we'd have:

in `fetchLatestProcessDetailsByQueryAction`
```
lastDeployedActionPerProcess <- fetchActionsOrEmpty(
  actionRepository
    .getLastActionPerProcess(ProcessActionState.FinishedStates, Some(Set(ScenarioActionName.Deploy)))
).map(_.filter { case (_, action) => action.state == ProcessActionState.Finished })
```

and in `fetchProcessDetailsForVersion`
```
// for last deploy action we are not interested in ExecutionFinished deploys - we don't want to show them in the history
lastDeployedActionData =
  actions.find(_.actionName == ScenarioActionName.Deploy).filter(_.state == ProcessActionState.Finished),
```

Reply:
Yeah, it looks like 3rd option (making both methods work the same, but otherwise than any of them) ;-) is correct. Now we need a test for that :)

Reply:
I miss one more thing. I understood @gskrobisz that he discovered that we have a bug in two places:
1. In the history panel - "DEPLOYED" badge is missing - I assume that `lastDeployedActionData` will fix it
2. In the status visible on the scenario list - I thought that `lastStateActionData` is used for that. Isn't it?

Reply:
> 1.In the history panel - "DEPLOYED" badge is missing - I assume that `lastDeployedActionData` will fix it
> 2.In the status visible on the scenario list - I thought that `lastStateActionData` is used for that. Isn't it?

1. yes, pretty sure that's correct
2. FE actually doesn't use `lastStateActionData` at all anywhere, but BE uses it to determine `ScenarioWithDetails.state` (in `DeploymentService.getProcessState`) which FE uses